aea/cli/generate_ipfs_hashes.py

```diff
@@ -0,0 +1,565 @@
+
+This script requires that you have IPFS installed:
+- https://docs.ipfs.io/guides/guides/install/
+"""
+import collections
+import csv
+import operator
+import os
+import pprint
+import re
+import shutil
+import signal
+import socket
+import subprocess  # nosec
+import sys
+import time
+import traceback
+from pathlib import Path
+from typing import Collection, Dict, List, Optional, Tuple, Type, cast
+
+import click
+import ipfshttpclient
+
+from aea.configurations.base import (
+    AgentConfig,
+    ConnectionConfig,
+    ContractConfig,
+    PackageConfiguration,
+    PackageType,
+    ProtocolConfig,
+    SkillConfig,
+    _compute_fingerprint,
+)
+from aea.configurations.constants import PACKAGE_TYPE_TO_CONFIG_FILE
+from aea.configurations.loader import ConfigLoaders
+from aea.helpers.yaml_utils import yaml_dump, yaml_dump_all
+
+
+CORE_PATH = Path("aea")
+TEST_PATH = Path("tests") / "data"
+PACKAGE_HASHES_PATH = "packages/hashes.csv"
+TEST_PACKAGE_HASHES_PATH = "tests/data/hashes.csv"
+
+type_to_class_config = {
+    PackageType.AGENT: AgentConfig,
+    PackageType.PROTOCOL: ProtocolConfig,
+    PackageType.CONNECTION: ConnectionConfig,
+    PackageType.SKILL: SkillConfig,
+    PackageType.CONTRACT: ContractConfig,
+}  # type: Dict[PackageType, Type[PackageConfiguration]]
+
+
+def _get_all_packages() -> List[Tuple[PackageType, Path]]:
+    """
+    Get all the hashable package of the repository.
+
+    In particular, get them from:
+    - aea/*
+    - packages/*
+    - tests/data/*
+
+    :return: pairs of (package-type, path-to-the-package)
+    """
+
+    def package_type_and_path(package_path: Path) -> Tuple[PackageType, Path]:
+        """Extract the package type from the path."""
+        item_type_plural = package_path.parent.name
+        item_type_singular = item_type_plural[:-1]
+        return PackageType(item_type_singular), package_path
+
+    CORE_PACKAGES = list(
+        map(
+            package_type_and_path,
+            [
+                CORE_PATH / "protocols" / "scaffold",
+                CORE_PATH / "connections" / "scaffold",
+                CORE_PATH / "contracts" / "scaffold",
+                CORE_PATH / "skills" / "scaffold",
+            ],
+        )
+    )
+
+    PACKAGES = list(
+        map(
+            package_type_and_path,
+            filter(operator.methodcaller("is_dir"), Path("packages").glob("*/*/*/")),
+        )
+    )
+
+    TEST_PACKAGES = [
+        (PackageType.AGENT, TEST_PATH / "dummy_aea"),
+        (PackageType.CONNECTION, TEST_PATH / "dummy_connection"),
+        (PackageType.CONTRACT, TEST_PATH / "dummy_contract"),
+        (PackageType.PROTOCOL, TEST_PATH / "generator" / "t_protocol"),
+        (PackageType.PROTOCOL, TEST_PATH / "generator" / "t_protocol_no_ct"),
+        (PackageType.SKILL, TEST_PATH / "dependencies_skill"),
+        (PackageType.SKILL, TEST_PATH / "exception_skill"),
+        (PackageType.SKILL, TEST_PATH / "dummy_skill"),
+    ]
```

Comment:
This makes zero sense. The framework should be usable anywhere. Hence, the path to tests folder needs to be provided as an arg and then the packages need to be detected.

Reply:
> This makes zero sense. The framework should be usable anywhere. Hence, the path to tests folder needs to be provided as an arg and then the packages need to be detected.

I've removed this and updated the method to retrieve package paths to find packages dynamically.